File: Frontend/src/app/models/user.model.ts
export interface LoginResponse {
  access_token: string;
  expires_at: string;
  employee_id: string;
  username: string;
  role: string;
  token_type: 'bearer';
}

export interface RegisterRequest {
  first_name: string;
  last_name: string;
  username: string;
  email: string;
  password: string;
  phone: string;
  department_id: string; 
}

export interface User {
  first_name: string;
  last_name: string;
  username: string;
  email: string;
  phone: string;
  employee_id: string; 
  role: string; 
  department_id: string;
  has_government_license: boolean;
  license_check_passed: boolean;
  license_file_url?: string | null;
  license_expiry_date: Date | null;
  is_blocked: boolean;
  block_expires_at?: Date | null; 
  block_reason?: string | null;
  is_unassigned_user?: boolean;
}

File: Backend/src/models/user_model.py
import uuid
from sqlalchemy import Column, String, Enum, ForeignKey , Boolean , Date
from sqlalchemy.dialects.postgresql import UUID
from src.models.base import Base
import enum
from sqlalchemy.orm import relationship

class UserRole(str, enum.Enum):
    anonymous = "anonymous"
    employee = 'employee'
    supervisor = 'supervisor'
    admin = 'admin'
    inspector = 'inspector'

class User(Base):
    __tablename__ = 'users'  

    
    first_name = Column(String, nullable=False)
    last_name = Column(String, nullable=False)
    username = Column(String, nullable=False, unique=True)
    phone = Column(String(20), nullable=True)
    email = Column(String, nullable=False, unique=True)
    employee_id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    password = Column(String, nullable=False)  # hashing afterwards
    role = Column(Enum(UserRole), nullable=False)
    department_id = Column(UUID(as_uuid=True), ForeignKey("departments.id"), nullable=False)
    has_government_license = Column(Boolean, default=False)
    license_file_url = Column(String, nullable=True)
    license_expiry_date = Column(Date, nullable=True)
    

    no_show_events = relationship("NoShowEvent", back_populates="user")
